safe-hts-precompile/SafeHTS.sol

```diff
@@ -0,0 +1,458 @@
+            abi.encodeWithSelector(IHederaTokenService.createNonFungibleTokenWithCustomFees.selector,
+            token, fixedFees, royaltyFees));
+        (responseCode, tokenAddress) =
+        success
+        ? abi.decode(result, (int32, address))
+        : (HederaResponseCodes.UNKNOWN, address(0));
+        require(responseCode == HederaResponseCodes.SUCCESS, "Safe create non fungible token with custom fees failed!");
+    }
+
+    function safeApprove(address token, address spender, uint256 amount) internal {
+        int32 responseCode;
+        (bool success, bytes memory result) = precompileAddress.call(
+            abi.encodeWithSelector(IHederaTokenService.approve.selector, token, spender, amount));
+        responseCode = success ? abi.decode(result, (int32)) : HederaResponseCodes.UNKNOWN;
+        require(responseCode == HederaResponseCodes.SUCCESS, "Safe approve failed!");
+    }
+
+    function safeAllowance(address token, address owner, address spender) internal
+    returns (uint256 allowance)
+    {
+        int32 responseCode;
+        (bool success, bytes memory result) = precompileAddress.call(
+            abi.encodeWithSelector(IHederaTokenService.allowance.selector, token, owner, spender));
+        (responseCode, allowance) = success ? abi.decode(result, (int32, uint256)) : (HederaResponseCodes.UNKNOWN, 0);
+        require(responseCode == HederaResponseCodes.SUCCESS, "Safe allowance failed!");
+    }
+
+    function safeApproveNFT(address token, address approved, int64 serialNumber) internal {
+        int32 responseCode;
+        (bool success, bytes memory result) = precompileAddress.call(
+            abi.encodeWithSelector(IHederaTokenService.approveNFT.selector, token, approved, serialNumber));
+        responseCode = success ? abi.decode(result, (int32)) : HederaResponseCodes.UNKNOWN;
+        require(responseCode == HederaResponseCodes.SUCCESS, "Safe NFT approve failed!");
+    }
+
+    function safeGetApproved(address token, int64 serialNumber) internal
+    returns (address approved)
+    {
+        int32 responseCode;
+        (bool success, bytes memory result) = precompileAddress.call(
+            abi.encodeWithSelector(IHederaTokenService.getApproved.selector, token, serialNumber));
+        (responseCode, approved) = success ? abi.decode(result, (int32, address)) : (HederaResponseCodes.UNKNOWN, address(0));
+        require(responseCode == HederaResponseCodes.SUCCESS, "Safe get approved failed!");
+    }
+
+    function safeSetApprovalForAll(address token, address operator, bool approved) internal {
+        int32 responseCode;
+        (bool success, bytes memory result) = precompileAddress.call(
+            abi.encodeWithSelector(IHederaTokenService.setApprovalForAll.selector, token, operator, approved));
+        responseCode = success ? abi.decode(result, (int32)) : HederaResponseCodes.UNKNOWN;
+        require(responseCode == HederaResponseCodes.SUCCESS, "Safe set token approval failed!");
```

Comment:
The error message could be changed to: "Safe set token approval for all failed!"